ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -61,99 +61,125 @@ def _generate_base_peer_group(self, pg_type: str, pg_name: str) -> dict:
+                peer_groups.append(
+                    {
+                        "name": self._peer_group_evpn_overlay_core,
+                        **self._generate_base_peer_group("evpn", "evpn_overlay_core"),
+                        "ebgp_multihop": self._evpn_ebgp_gateway_multihop,
+                    }
+                )
 
         elif self._overlay_routing_protocol == "ibgp":
             if self._overlay_mpls is True:
                 # MPLS OVERLAY peer group
                 local_as = str(_as) if (_as := get(self._hostvars, "switch.ipvpn_gateway.local_as")) is not None else None
-                peer_groups[self._peer_group_mpls_overlay_peers] = self._generate_base_peer_group("mpls", "mpls_overlay_peers")
-                peer_groups[self._peer_group_mpls_overlay_peers].update(
-                    {
-                        "local_as": local_as,
-                        "remote_as": self._bgp_as,
-                    }
-                )
+                mpls_peer_group = {
+                    "name": self._peer_group_mpls_overlay_peers,
+                    **self._generate_base_peer_group("mpls", "mpls_overlay_peers"),
+                    "local_as": local_as,
+                    "remote_as": self._bgp_as,
+                }
 
                 if self._mpls_overlay_role == "server" or (self._evpn_role == "server" and get(self._hostvars, "overlay.evpn_mpls") is True):
-                    peer_groups[self._peer_group_mpls_overlay_peers]["route_reflector_client"] = True
+                    mpls_peer_group["route_reflector_client"] = True
+
+                peer_groups.append(mpls_peer_group)
 
             if self._overlay_evpn_vxlan is True:
                 # EVPN OVERLAY peer group - also in EBGP..
-                peer_groups[self._peer_group_evpn_overlay_peers] = self._generate_base_peer_group("evpn", "evpn_overlay_peers")
-                peer_groups[self._peer_group_evpn_overlay_peers]["remote_as"] = self._bgp_as
+                ebgp_peer_group = {
+                    "name": self._peer_group_evpn_overlay_peers,
+                    **self._generate_base_peer_group("evpn", "evpn_overlay_peers"),
+                    "remote_as": self._bgp_as,
+                }
+
                 if self._evpn_role == "server":
-                    peer_groups[self._peer_group_evpn_overlay_peers]["route_reflector_client"] = True
+                    ebgp_peer_group["route_reflector_client"] = True
+
+                peer_groups.append(ebgp_peer_group)
 
             if self._is_mpls_server is True:
-                peer_groups[self._peer_group_rr_overlay_peers] = self._generate_base_peer_group("mpls", "rr_overlay_peers")
-                peer_groups[self._peer_group_rr_overlay_peers]["remote_as"] = self._bgp_as
+                peer_groups.append(
+                    {"name": self._peer_group_rr_overlay_peers, **self._generate_base_peer_group("mpls", "rr_overlay_peers"), "remote_as": self._bgp_as}
```

Comment:
if we added a `name` argument to `_generate_base_peer_group` you could simplify a lot here.